cumulocity-blocks/CreateMultiMeasurement.mon

```diff
@@ -0,0 +1,153 @@
+using apama.analyticsbuilder.cumulocity.CumulocityOutputHandler;
+using apama.analyticsbuilder.Promise;
+using apama.analyticsbuilder.Value;
+
+event CreateMultiMeasurement_$Parameters {
+	/**
+	 * Device or Trigger Device.
+	 *
+	 * The device (or for models handling groups, Trigger Device) to which the measurement is to be sent.
+	 *
+	 * The model editor uses the current device name. This is mapped internally to the device identifier.
+	 * @$semanticType c8y_deviceIdOrCurrentDevice
+	 */
+	any deviceId;
+
+	/**
+	 * Measurement Type.
+	 *
+	 * The type of the measurement.
+	 */
+	string measurementType;
+
+	/** Validate parameters */
+	action $validate() {
+		switch(deviceId) {
+			case string: { BlockBase.throwsOnEmpty(deviceId, "deviceId", self); }
+			default: {
+				if deviceId = any() or deviceId.getEntry("currentDevice") != <any> true {
+					throw Exception("Expected a device Id or a currentDevice object", "IllegalArgumentException"); // NON-L10N-DEV
+				}
+			}
+		}
+		BlockBase.throwsOnEmpty(measurementType, "measurementType", self);
+	}
+}
+
+/**
+ * Complex Measurement Output.
+ *
+ * Creates a new complex <tt>Measurement</tt> object for the associated device or the triggering device.
+ *
+ * This block sends a new measurement to Cumulocity IoT containing multiple fragments and series. 
+ * The list of fragments and series is included as properties on the Value input. The measurement
+ * is sent to the current device or the device specified.
+ *
+ * Non-finite values are ignored.
+ *
+ * @$blockCategory Output
+ * @$producesOutput
+ * @$blockType c8y_Output
+ * @$derivedName $measurementType Measurement
+ * @$titleIsDerived true
+ */
+event CreateMultiMeasurement {
+	BlockBase $base;
+	CreateMultiMeasurement_$Parameters $parameters;
+	/** Sends to the 'current' device from the activation's partition: */
+	CumulocityOutputHandler outputHandler;
+	
+	action $validate() returns Promise {
+		dictionary<string, any> fields := {"type":<any>$parameters.measurementType};
+	    // Creating an CumulocityOutputParams object
+	    CumulocityOutputParams params := CumulocityOutputParams.forSyncEventType($parameters.deviceId, 
+	                                            self, Measurement.getName(), fields);
+	    // Declare the output stream and provide callback action to save the output handler object.
+	    // Return the Promise returned from the declare call.
+	    return params.declare(handlerCreated);
+	}
+	/** Callback action to receive and save CumulocityOutputHandler object.
+	 * @param cumulocityOutputHandler the handler object
+	 */
+	action handlerCreated(CumulocityOutputHandler cumulocityOutputHandler) {
+	    outputHandler := cumulocityOutputHandler;
+	}
+
+	
+	/**
+	 * Process an activation.
+	 * @param $input_value Properties containing the values to be included in the measurement. Keys are "fragment.series" and the values are either floats or objects with "value" and "unit" (optional).
+	 * @$inputName value Value
```

Comment:
Ok to call this input "value" even though it is a pulse?

Reply:
Yes, it should not be a problem.